### PR TITLE
Refresh improvment

### PR DESCRIPTION
**Pull request by slawekjaranowski**
I hope this help some of refresh issues.

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
#158 is still needed

**Comment by digitaldan** [comment]
Hi again, before I pull down and test this, what is the refresh issue that this was solving?  Is there a bug opened that this solves?  Thanks.

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetListFragment.java [hunk 1]
```diff
@@ -412,49 +412,49 @@ public void processContent(String responseString, boolean longPolling) {
...
         // If openHAB verion = 1 get page from XML
-        if (mActivity.getOpenHABVersion() == 1) {
-            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
-            try {
-                DocumentBuilder builder = dbf.newDocumentBuilder();
-                Document document = builder.parse(new InputSource(new StringReader(responseString)));
-                if (document != null) {
-                    Node rootNode = document.getFirstChild();
-                    openHABWidgetDataSource.setSourceNode(rootNode);
+        if (responseString != null && responseString.length() > 0) {
```
Comment (digitaldan) [inline mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetListFragment.java]: If the response string is null or empty what happens?  Line 433 handles the case where the xml document is null and I believe tries to reconnect. should something similar happen here as well?

File: mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetListFragment.java [hunk 2]
```diff
@@ -412,49 +412,49 @@ public void processContent(String responseString, boolean longPolling) {
...
+            if (mActivity.getOpenHABVersion() == 1) {
+                DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+                try {
+                    DocumentBuilder builder = dbf.newDocumentBuilder();
+                    Document document = builder.parse(new InputSource(new StringReader(responseString)));
```
Comment (slawekjaranowski) [inline mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetListFragment.java]: When we have empty response "builder.parse" throws exception, so next statement isn't executed

**Comment by slawekjaranowski** [comment]
I simplify my change ... and add some comments in code.

There is no issue for this, when I was testing my other change I had some stack trace form time to time  - caused by empty response from OH.

In this case I think it is ok to stop processing of refresh page.
Maybe it is issue somewhere else.

**Comment by digitaldan** [comment]
Thanks @slawekjaranowski , I'll pull this down in a bit and test.

**Pull request merged**